src/condor_scripts/condor_upgrade_check

```diff
@@ -507,16 +507,59 @@ def check_gpu_requirements():
         format_print("Failed to locate local schedd.", offset=8, err=True)
 
 
+def warn_dagman_env():
+    """Warn about DAGMan cleansing its environment"""
+    if "SCHEDD" not in htcondor.param["DAEMON_LIST"]:
+        return 0
+    htc_version = htcondor.param["CONDOR_VERSION"].split(".")
+    if int(htc_version[0]) < 10 or (
+        int(htc_version[0]) == 10 and int(htc_version[1]) < 4
+    ):
+        format_print("Warning:", newline=True)
+        format_print("DAGMan no longer copies the entire enviroment that it", offset=4)
+        format_print("was submitted from. This will affect anything that", offset=4)
+        format_print("depends on environment variables, including jobs and", offset=4)
```

Comment:
Some nitpicking, but "anything" is a little broad and scary here.
```suggestion
        format_print("DAGMan no longer copies the entire environment from", offset=4)
        format_print("which it was submitted. This will affect any DAG items that", offset=4)
        format_print("depend on environment variables, including jobs and", offset=4)
```